sway-core/src/asm_generation/fuel/register_allocator.rs

```diff
@@ -374,8 +426,54 @@ pub(crate) fn coalesce_registers(
     for new_op in &mut reduced_ops {
         *new_op = new_op.update_register(&final_reg_to_reg_map);
     }
+    for new_live_out in &mut reduced_live_out {
+        for (old, &new) in &final_reg_to_reg_map {
+            if new_live_out.remove(old) {
+                new_live_out.insert(new.clone());
+            }
+        }
+    }
+
+    (reduced_ops, reduced_live_out)
+}
+
+// For every virtual register, compute its (def points, use points).
+fn compute_def_use_points(ops: &[Op]) -> FxHashMap<VirtualRegister, (Vec<usize>, Vec<usize>)> {
+    let mut res: FxHashMap<VirtualRegister, (Vec<usize>, Vec<usize>)> = FxHashMap::default();
+    for (idx, op) in ops.iter().enumerate() {
+        let mut op_use = op.use_registers();
+        let mut op_def = op.def_registers();
+        op_use.retain(|&reg| matches!(reg, VirtualRegister::Virtual(_)));
```

Comment:
Is it worth it introducing an `is_virtual` method to `VirtualRegister` instead of repeating macro use here? Could improve readability.